jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/core/Float4VectorTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.salesforce.datacloud.jdbc.core;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import com.salesforce.datacloud.jdbc.hyper.HyperTestBase;
+import io.grpc.ManagedChannelBuilder;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(HyperTestBase.class)
+public class Float4VectorTest {
+
+    @Test
+    public void testFloat4VectorQuery() throws SQLException {
+        Properties properties = new Properties();
+        ManagedChannelBuilder<?> channel = ManagedChannelBuilder.forAddress(
+                        "127.0.0.1", HyperTestBase.getInstancePort())
+                .usePlaintext();
+
+        // Expected values from generate_series(1, 10, 0.5)
+        List<Float> expectedValues = new ArrayList<>();
+        for (float i = 1.0f; i <= 10.0f; i += 0.5f) {
+            expectedValues.add(i);
+        }
+
+        try (DataCloudConnection conn = DataCloudConnection.fromChannel(channel, properties)) {
+            try (Statement stmt = conn.createStatement()) {
```

Comment:
for this (managed channel and fromChannel) setup you can just use `assertWithStatement()` from HyperTestBase - we show the managed channel bit in those sample tests to be pedantic, but here we should try for brevity so the logic of the test is more clear.